op-node/p2p/pings.go

```diff
@@ -41,7 +41,13 @@ type PingService struct {
 	wg sync.WaitGroup
 }
 
-func NewPingService(log log.Logger, ping PingFn, peers PeersFn, clock clock.Clock) *PingService {
+func NewPingService(
+	log log.Logger,
+	ping PingFn,
+	peers PeersFn,
+	clock clock.Clock,
+	trace func(work string),
+) *PingService {
```

Comment:
The `trace` field is only used in tests, in `op-node/p2p/pings_test.go`, so I think it's better to keep the old code to shield the `trace` from the public api and only set the field in the test manually. That's better than passing a `nil` in the prod code at `op-node/p2p/host.go`.

Reply:
This fixes an actual race because otherwise pingPeersBackground starts before the trace is set. I'll make a private constructor for the tests with the trace parameter.

Reply:
Fixed